### PR TITLE
fix(asgi): ensure error tags are set on request spans

### PR DESCRIPTION
**Pull request by mabdinur**
If an exception is raised in the ddtrace ASGI middleware the `<asgi_framework_name>.request` span will be finished before the ``error.message`` and ``error.stack`` tags are set on the span. Currently these tags are not submitted to the agent.

Reproduction:

1. Install `fastapi` and `ddtrace==1.9.0`
2. Create fastapi route that raises an Exception. Ex:
    ```python
      from fastapi import FastAPI
      app = FastAPI()

      @app.get("/abc")
      def hello() -> str:
          raise ValueError('error')
          return "ok"
    ```
3. Run fastapi app with ddtrace-run
4. Result:
     - Screen Shot 2023-05-09 at 3 13 05 PM

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] OPTIONAL: PR description includes explicit acknowledgement of the performance implications of the change as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: ddtrace/contrib/asgi/middleware.py [hunk 1]
```diff
@@ -259,9 +247,6 @@ async def wrapped_blocked_send(message):
                 self.handle_exception_span(exc, span)
                 reraise(exc_type, exc_val, exc_tb)
             finally:
-                try:
-                    del scope["datadog"]["request_span"]
-                except KeyError:
-                    pass
-
+                if span in scope["datadog"]["request_spans"]:
```
Comment (ZStriker19) [inline ddtrace/contrib/asgi/middleware.py:250]: Could you explain why we need to change this as well?

**Review by ZStriker19 — COMMENTED** [review]
Looks good, just one question

File: ddtrace/contrib/asgi/middleware.py (hunk 1 above)
Comment (emmettbutler) [reply ddtrace/contrib/asgi/middleware.py:250]: This change looks to me like a less-verbose way of doing the same `del` logic that was on lines 262-265.

**Review by emmettbutler — APPROVED** [review]

File: ddtrace/contrib/asgi/middleware.py (hunk 1 above)
Comment (mabdinur) [reply ddtrace/contrib/asgi/middleware.py:250]: It's a bit subtle but this is actually a "bug fix". scope["datadog"]["request_span"] is never set, we actually set scope["datadog"]["request_span**s**"].  This line always raises a KeyError. The impact of this bug is pretty low. We'll just hold more spans in memory than we need to. I can move this fix to it's own PR.

**Review by juanjux — APPROVED** [review]

**Comment by victorcelerier** [comment]
Hi everyone, following this commit if I'm not wrong, it means we will include background task in web requests span again right ? It was removed inside https://github.com/DataDog/dd-trace-py/commit/5b97489e2b073fa773819904cdef26981a5a28df (asgi, starlette, fastapi: Exclude background tasks duration from web request span )

**Comment by Yun-Kim** [comment]
@mabdinur just for clarity for future reviewers, can we add what the initial problem was and how we fixed it in the PR description? Thanks!

**Pull request merged**